packages/ERTP/src/mathHelpers/natListMathHelpers.js

```diff
@@ -34,7 +36,11 @@ const natListMathHelpers = harden({
   doSubtract: (left, right) => {
     const leftSet = new Set(left);
     const remove = elem => leftSet.delete(elem);
-    right.forEach(remove);
+    const allRemovedCorrectly = right.every(remove);
+    assert(
+      allRemovedCorrectly,
+      `the elements in right (${right}) could not be removed from left (${left})`,
```

Comment:
While literally true, this is likely obscuring the bug that caused the failure to remove: that an element in `right` was not present in `left`. Since this is exactly the symptom of confusing `subtract` on sets with set subtraction, we should be clear.

Reply:
Ah, are you saying just that the error message needs to change, or that something else should be done? (I see I need to add `details` here)

Reply:
```js
details`the elements in right (${right}) were not present in left (${left})`
```
sounds like all elements rather than some elements.

Reply:
ah, ok I'll fix!